test/concuerror_tests.erl

```diff
@@ -42,33 +42,88 @@ optvar_read_test() ->
         cleanup()
     end.
 
+optvar_zombie_test() ->
+    init(),
+    Var1 = {foo, bar},
+    Var2 = {bar, foo},
+    Val1 = 1,
+    Val2 = 1,
+    try
+        timeout = optvar:read(Var1, 0),
+        timeout = optvar:read(Var2, 0),
+        [{_, {unset, Pid}}] = ets:lookup(optvar_status_tab, Var1),
+        %% Create a zombie entry in the status table:
+        MRefWaker = monitor(process, Pid),
+        exit(Pid, kill),
+        receive
+            {'DOWN', MRefWaker, _, _, _} -> ok
+        end,
+        %% Now spawn workers that will encounter the zombie:
+        Setter1 = spawn(fun() ->
+                                ?assertMatch(ok, optvar:set(Var1, Val1))
+                        end),
+        Setter2 = spawn(fun() ->
+                                ?assertMatch(ok, optvar:set(Var2, Val2))
+                        end),
+        Reader = spawn(fun() ->
+                               ?assertEqual(Val1, optvar:read(Var1)),
+                               ?assertEqual(Val2, optvar:read(Var2))
+                       end),
+        %% Regardless, all the processes should avoid deadloop:
+        wait_pids([Reader, Setter1, Setter2])
+    after
+        cleanup()
+    end.
+
+optvar_set_unset_race_test() ->
+    init(),
+    Var = {foo, bar},
+    Val = 1,
+    try
+        Setter = spawn(fun() ->
+                               ?assertMatch(ok, optvar:set(Var, Val))
+                       end),
+        Unsetter = spawn(fun() ->
+                                 ?assertMatch(ok, optvar:unset(Var))
+                         end),
+        Reader1 = spawn(fun() ->
+                                ?assertEqual(Val, optvar:read(Var))
+                        end),
+        Reader2 = spawn(fun() ->
+                                ?assertEqual(Val, optvar:read(Var))
+                        end),
+        %% Wait until the variable is unset:
+        wait_pids([Unsetter]),
+        ?assertMatch(ok, optvar:set(Var, Val)),
+        %% Now when the variable is set by either process, both reader
+        %% processes should be able to terminate:
+        wait_pids([Setter, Reader1, Reader2])
+    after
+        cleanup()
+    end.
```

Comment:
Maybe we could repeat this test but also attempting to stop/kill the supervisor which is the ETS owner, so we could approximate the application master killing the `optvar` app?  Would this make sense?  :thinking:

Reply:
So, the original problem with application master killing the waker process was due to the group leader belonging to the emqx app, not the optvar app?